src/client/rewards/MatchBot/MatchBot.js

```diff
@@ -1,16 +1,74 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import PropTypes from 'prop-types';
-import { Link } from 'react-router-dom';
+import { connect } from 'react-redux';
 import { injectIntl } from 'react-intl';
-import _ from 'lodash';
+import { isEmpty } from 'lodash';
+import { Button, message, Modal, Slider } from 'antd';
+import { setMatchBotVotingPower } from '../rewardsActions';
+import CreateRule from './CreateRule/CreateRule';
+import { getMatchBotRules } from '../../../waivioApi/ApiClient';
 import MatchBotTable from './MatchBotTable/MatchBotTable';
 import './MatchBot.less';
 
-const MatchBot = ({ intl }) => {
-  // Mock
-  const sponsors = [
-    { id: 1, isActive: true, name: 'sponsor_1', upvote: 100, action: 'edit', notes: 'some note' },
-  ];
+const MatchBot = ({ intl, userName, ...props }) => {
+  const [modalVisible, setModalVisible] = useState(false);
+  const [voteModalVisible, setVoteModalVisible] = useState(false);
+  const [isLoading, setLoaded] = useState(false);
+  const [sliderValue, setSliderValue] = useState(100);
+  const [editRule, setEditRule] = useState({});
+  const [rules, setRules] = useState({ results: [] });
+  const [minVotingPower, setMinVotingPower] = useState(0);
+  const maxRulesLimit = 25;
+  const isOverRules = rules.results.length >= maxRulesLimit;
+  const marks = {
+    1: '1%',
+    25: '25%',
+    50: '50%',
+    75: '75%',
+    100: '100%',
+  };
+
+  useEffect(() => {
+    getMatchBotRules(userName).then(data => {
+      setRules(data);
+      setMinVotingPower(80);
+    });
+  }, []);
```

Comment:
Check number of function's calls. Maybe you need to add condition for calling the effect